lib/compilers/rust.ts

```diff
@@ -138,6 +138,8 @@ export class RustCompiler extends BaseCompiler {
             if (this.linker) {
                 options = options.concat(`-Clinker=${this.linker}`);
             }
+        } else if (filters.binaryobject) {
+            options = options.concat(['--crate-type', 'lib']);
```

Comment:
nice - looks like this teaches Rust a new trick too?

Reply:
Hmmm, have to double check this one. I decided to stick to gcc/clang for this PR, but this one survived. I'll test that and maybe remove it

Reply:
So I checked this is working as expected locally. Not really sure how to test something that pulls another crate...